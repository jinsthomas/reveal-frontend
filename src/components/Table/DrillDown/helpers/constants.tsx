/** Constants */
import React from 'react';

export const ID = 'id';
export type ID = typeof ID;

export const PARENT_ID = 'parent_id';
export type PARENT_ID = typeof PARENT_ID;

export const ROOT_PARENT_ID = null;
export type ROOT_PARENT_ID = typeof ROOT_PARENT_ID;

export const LINKER_ITEM_CSS_CLASS = 'dd-linker-item';
export type LINKER_ITEM_CSS_CLASS = typeof LINKER_ITEM_CSS_CLASS;

export const CLICKABLE_CSS_CLASS = 'dd-linker-item dd-clickable';
export type CLICKABLE_CSS_CLASS = typeof CLICKABLE_CSS_CLASS;

export const CARET_CSS_CLASS = 'dd-caret';
export type CARET_CSS_CLASS = typeof CARET_CSS_CLASS;

export const CARET_SPAN = <span className={CARET_CSS_CLASS}>&nbsp;&#9660;</span>;
export type CARET_SPAN = typeof CARET_SPAN;

export const DEFAULT_ROW_HEIGHT = '2em';
export type DEFAULT_ROW_HEIGHT = typeof DEFAULT_ROW_HEIGHT;
