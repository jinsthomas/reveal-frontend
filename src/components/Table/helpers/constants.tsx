/** Constants */
import React from 'react';
export const ID = 'id';
export type ID = typeof ID;

export const PARENT_ID = 'parent_id';
export type PARENT_ID = typeof PARENT_ID;

export const ROOT_PARENT_ID = null;
export type ROOT_PARENT_ID = typeof ROOT_PARENT_ID;

export const LINKER_ITEM_CSS_CLASS = 'dd-linker-item';
export type LINKER_ITEM_CSS_CLASS = typeof LINKER_ITEM_CSS_CLASS;

export const CLICKABLE_CSS_CLASS = 'dd-linker-item dd-clickable';
export type CLICKABLE_CSS_CLASS = typeof CLICKABLE_CSS_CLASS;

export const CARET_CSS_CLASS = 'dd-caret';
export type CARET_CSS_CLASS = typeof CARET_CSS_CLASS;

export const CARET = <span className={CARET_CSS_CLASS}>&nbsp;&#9660;</span>;
export type CARET = typeof CARET;

export const ROWS_TO_DISPLAY = 'Rows to display';
export type ROWS_TO_DISPLAY = typeof ROWS_TO_DISPLAY;

export const PAGE = 'page';
export type PAGE = typeof PAGE;

export const OF = 'of';
export type OF = typeof OF;
