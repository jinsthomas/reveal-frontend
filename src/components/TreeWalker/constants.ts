export const ACTIVE = 'Active';
export const FIND_BY_ID = 'location/findByJurisdictionIds';
export const FIND_BY_PROPERTIES = 'location/findByProperties';
export const LOCATION = 'location';
export const COULDNT_LOAD_PARENTS = 'Could not load parents';
