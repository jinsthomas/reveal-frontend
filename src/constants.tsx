// env vars
export const WEBSITE_NAME = process.env.REACT_APP_WEBSITE_NAME;
export type WEBSITE_NAME = typeof WEBSITE_NAME;

// strings
export const LOCATION = 'Location';
export type LOCATION = typeof LOCATION;
export const PROVINCE = 'Province';
export type PROVINCE = typeof PROVINCE;
export const HOME = 'Home';
export type HOME = typeof HOME;
export const FOCUS_INVESTIGATION = 'Focus Investigation';
export type FOCUS_INVESTIGATION = typeof FOCUS_INVESTIGATION;
export const FOCUS_INVESTIGATIONS = 'Focus Investigations';
export type FOCUS_INVESTIGATIONS = typeof FOCUS_INVESTIGATIONS;

// internal urls
export const HOME_URL = '/';
export type HOME_URL = typeof HOME_URL;
export const IRS_URL = '/irs';
export type IRS_URL = typeof IRS_URL;
export const FI_URL = '/focus-investigation';
export type FI_URL = typeof FI_URL;
export const FI_HISTORICAL_URL = '/focus-investigation/historical';
export type FI_HISTORICAL_URL = typeof FI_HISTORICAL_URL;

// colors
export const GREEN = 'Green';
export type GREEN = typeof GREEN;
export const YELLOW = 'Yellow';
export type YELLOW = typeof YELLOW;
export const ORANGE = 'Orange';
export type ORANGE = typeof ORANGE;
export const RED = 'Red';
export type RED = typeof RED;

// thresholds
export const GREEN_THRESHOLD = 0.9;
export type GREEN_THRESHOLD = typeof GREEN_THRESHOLD;
export const YELLOW_THRESHOLD = 0.2;
export type YELLOW_THRESHOLD = typeof YELLOW_THRESHOLD;
export const ORANGE_THRESHOLD = 0.8;
export type ORANGE_THRESHOLD = typeof ORANGE_THRESHOLD;

// 1-3-7 thresholds
export const THRESHOLD_137 = 0;
