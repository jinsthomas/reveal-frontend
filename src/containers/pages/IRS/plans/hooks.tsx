import { SupersetFormData } from '@onaio/superset-connector/dist/types';
import React, { useEffect, useState } from 'react';
import { ActionCreator } from 'redux';
import { SUPERSET_IRS_REPORTING_PLANS_SLICE } from '../../../../configs/env';
import { displayError } from '../../../../helpers/errors';
import supersetFetch from '../../../../services/superset';

interface AsyncSupersetPlans<TData, TAction> {
  superset: typeof supersetFetch;
  data: TData[];
  fetchPlans: ActionCreator<TAction>;
  supersetSlice: string;
  supersetOptions: SupersetFormData | null;
}

export const useAsyncSupersetPlans = <TData, TAction>({
  superset = supersetFetch,
  data = [],
  fetchPlans,
  supersetSlice = SUPERSET_IRS_REPORTING_PLANS_SLICE,
  supersetOptions = null,
}: AsyncSupersetPlans<TData, TAction>) => {
  const [loading, setLoading] = useState<boolean>(false);
  const mounted = React.useRef<boolean>(false);

  /** async function to load the data */
  async function loadData() {
    try {
      setLoading(data.length === 0); // only set loading when there are no plans
      const asyncOperation = supersetOptions
        ? superset(supersetSlice, supersetOptions)
        : superset(supersetSlice);

      await asyncOperation.then((result: TData[]) => fetchPlans(result));
    } catch (e) {
      throw e;
    } finally {
      if (mounted) {
        setLoading(false);
      }
    }
  }

  useEffect(() => {
    mounted.current = true;
    loadData().catch(error => displayError(error));
    return () => {
      mounted.current = false;
    };
  }, []);

  return loading;
};
