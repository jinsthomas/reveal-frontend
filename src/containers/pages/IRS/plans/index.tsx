import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import { SUPERSET_IRS_REPORTING_PLANS_SLICE } from '../../../../configs/env';
import { IRS_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_IRS_PLAN_URL } from '../../../../constants';
import { getQueryParams, RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import IRSPlansReducer, {
  fetchIRSPlans,
  GenericPlan,
  makeIRSPlansArraySelector,
  reducerName as IRSPlansReducerName,
} from '../../../../store/ducks/generic/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the IRS plan definitions reducer */
reducerRegistry.register(IRSPlansReducerName, IRSPlansReducer);

/** interface for IRS PlanList props */
interface PlanListProps extends GenericPlanListProps {
  ownProps: any;
}

/** Simple component that loads a privew list of IRS plans */
const IRSPlansList = (props: PlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for IRSPlansList */
const defaultProps: PlanListProps = {
  fetchPlans: fetchIRSPlans,
  ownProps: {},
  pageTitle: IRS_PLANS,
  pageUrl: REPORT_IRS_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_IRS_REPORTING_PLANS_SLICE,
};

IRSPlansList.defaultProps = defaultProps;

export { IRSPlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const plans = makeIRSPlansArraySelector()(state, { plan_title: searchedTitle });

  return {
    ...ownProps,
    plans,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: fetchIRSPlans };

/** Connected ActiveFI component */
const ConnectedIRSPlansList = connect(mapStateToProps, mapDispatchToProps)(IRSPlansList);

export default ConnectedIRSPlansList;
