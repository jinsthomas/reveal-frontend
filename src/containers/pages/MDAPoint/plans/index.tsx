import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import { SUPERSET_MDA_POINT_REPORTING_PLANS_SLICE } from '../../../../configs/env';
import { MDA_POINT_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_MDA_POINT_PLAN_URL } from '../../../../constants';
import { getQueryParams } from '../../../../helpers/utils';
import { RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import MDAPointPlansReducer, {
  fetchMDAPointPlans,
  makeMDAPointPlansArraySelector,
  MDAPointPlan,
  reducerName as MDAPointPlansReducerName,
} from '../../../../store/ducks/generic/MDAPointPlans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the MDA plan definitions reducer */
reducerRegistry.register(MDAPointPlansReducerName, MDAPointPlansReducer);

// /** interface for MDA PlanList props */
interface PlanListProps extends GenericPlanListProps {
  ownProps: any;
}

/** Simple component that loads a preview list of MDA plans */
const MDAPointPlansList = (props: PlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for MDAPointPlansList */
const defaultProps: PlanListProps = {
  fetchPlans: fetchMDAPointPlans,
  ownProps: {},
  pageTitle: MDA_POINT_PLANS,
  pageUrl: REPORT_MDA_POINT_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_MDA_POINT_REPORTING_PLANS_SLICE,
};

MDAPointPlansList.defaultProps = defaultProps;

export { MDAPointPlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: MDAPointPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const MDAPointPlansArray = makeMDAPointPlansArraySelector()(state, { plan_title: searchedTitle });

  return {
    ...ownProps,
    plans: MDAPointPlansArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: fetchMDAPointPlans };

/** Connected ActiveFI component */
const ConnectedMDAPointPlansList = connect(mapStateToProps, mapDispatchToProps)(MDAPointPlansList);

export default ConnectedMDAPointPlansList;
