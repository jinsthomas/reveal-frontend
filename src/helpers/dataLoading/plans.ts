import { toast } from 'react-toastify';
import { ActionCreator, AnyAction } from 'redux';
import { OPENSRP_PLANS_BY_USER_FILTER } from '../../constants';
import { OpenSRPService } from '../../services/opensrp';
import store from '../../store';
import { fetchPlansByUser, FetchPlansByUserAction } from '../../store/ducks/opensrp/planIdsByUser';
import { growl } from '../utils';

/** find plans that the given user has access to
 * @param {string} userName - openMRSUserName
 * @param {ActionCreator<FetchPlansByUserAction>} actionCreator - action creator for adding practitioners to store
 * @param {typeof OpenSRPService} service -  the OpenSRP service
 * @param {ActionCreator<AnyAction>} responseActionCreator - optionalActionCreator to dispatch the planData in response
 */
export async function loadPlansByUserFilter<T>(
  userName: string,
  actionCreator: ActionCreator<FetchPlansByUserAction> = fetchPlansByUser,
  service: typeof OpenSRPService = OpenSRPService,
  responseActionCreator?: ActionCreator<AnyAction>
) {
  const serve = new service(`${OPENSRP_PLANS_BY_USER_FILTER}/${userName}`);
  serve
    .list()
    .then((response: T[]) => {
      store.dispatch(actionCreator(response, userName));
      if (responseActionCreator) {
        store.dispatch(responseActionCreator(response));
      }
    })
    .catch((err: Error) => {
      growl(err.message, { type: toast.TYPE.ERROR });
    });
}
