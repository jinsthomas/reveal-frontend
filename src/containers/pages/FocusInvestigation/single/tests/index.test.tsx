import { library } from '@fortawesome/fontawesome-svg-core';
import { faExternalLinkSquareAlt } from '@fortawesome/free-solid-svg-icons';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Helmet } from 'react-helmet';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { FI_SINGLE_URL } from '../../../../../constants';
import store from '../../../../../store';
import { fetchGoals } from '../../../../../store/ducks/goals';
import { fetchJurisdictions } from '../../../../../store/ducks/jurisdictions';
import { fetchPlans, Plan } from '../../../../../store/ducks/plans';
import * as fixtures from '../../../../../store/ducks/tests/fixtures';
import ConnectedSingleFI, { SingleFI } from '../../single';

jest.mock('../../../../../components/GisidaWrapper', () => {
  const GisidaWrapperMock = () => <div>I love oov</div>;
  return GisidaWrapperMock;
});

jest.mock('../../../../../configs/env');
jest.mock('@onaio/superset-connector', () => {
  // tslint:disable-next-line: label-position
  const superset = {
    getFormData: () => ({}),
  };
  return superset;
});
const history = createBrowserHistory();
library.add(faExternalLinkSquareAlt);
describe('containers/pages/SingleFI', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    const mock: any = jest.fn();
    const props = {
      completeReactivePlansArray: [fixtures.completeReactivePlan],
      completeRoutinePlansArray: [fixtures.completeRoutinePlan],
      currentReactivePlansArray: [fixtures.plan2],
      currentRoutinePlansArray: [fixtures.plan1],
      goalsArray: fixtures.plan1Goals,
      history,
      jurisdiction: fixtures.jurisdictions[0],
      location: mock,
      match: {
        isExact: true,
        params: { id: fixtures.plan1.id },
        path: `${FI_SINGLE_URL}/:id`,
        url: `${FI_SINGLE_URL}/13`,
      },
      planById: fixtures.plan1 as Plan,
      plansIdArray: fixtures.plansIdArray,
    };
    shallow(
      <Router history={history}>
        <SingleFI {...props} />
      </Router>
    );
  });

  it('renders SingleFI correctly', () => {
    const mock: any = jest.fn();
    const supersetServiceMock: any = jest.fn(async () => []);
    const props = {
      completeReactivePlansArray: [fixtures.completeReactivePlan],
      completeRoutinePlansArray: [fixtures.completeRoutinePlan],
      currentReactivePlansArray: [fixtures.plan2],
      currentRoutinePlansArray: [fixtures.plan1],
      goalsArray: fixtures.plan1Goals,
      history,
      jurisdiction: fixtures.jurisdictions[0],
      location: mock,
      match: {
        isExact: true,
        params: { id: fixtures.plan1.id },
        path: `${FI_SINGLE_URL}/:id`,
        url: `${FI_SINGLE_URL}/16`,
      },
      planById: fixtures.plan1 as Plan,
      plansIdArray: fixtures.plansIdArray,
      supersetService: supersetServiceMock,
      // tslint:disable-next-line: object-literal-sort-keys
      fetchPlansActionCreator: jest.fn(),
      fetchGoalsActionCreator: jest.fn(),
      fetchJurisdictionActionCreator: jest.fn(),
    };
    const wrapper = mount(
      <Router history={history}>
        <SingleFI {...props} />
      </Router>
    );
    // check that the documents title was changed correctly
    const helmet = Helmet.peek();
    expect(helmet.title).toEqual('Focus Investigations in NVI_439');

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('GisidaWrapperMock').props()).toMatchSnapshot();
    expect(wrapper.find('HeaderBreadcrumb').length).toEqual(1);
    wrapper.unmount();
  });

  it('renders SingleFI correctly for null plan jurisdction id', () => {
    const mock: any = jest.fn();
    const supersetServiceMock: any = jest.fn(async () => []);
    const props = {
      completeReactivePlansArray: [],
      completeRoutinePlansArray: [],
      currentReactivePlansArray: [],
      currentRoutinePlansArray: [],
      goalsArray: [],
      history,
      jurisdiction: fixtures.jurisdictions[0],
      location: mock,
      match: {
        isExact: true,
        params: { id: fixtures.plan5.id },
        path: `${FI_SINGLE_URL}/:id`,
        url: `${FI_SINGLE_URL}/16`,
      },
      planById: fixtures.plan5,
      plansArray: [fixtures.plan5],
      plansIdArray: [fixtures.plan5.id],
      supersetService: supersetServiceMock,
      // tslint:disable-next-line: object-literal-sort-keys
      fetchPlansActionCreator: jest.fn(),
      fetchGoalsActionCreator: jest.fn(),
      fetchJurisdictionActionCreator: jest.fn(),
    };
    const wrapper = mount(
      <Router history={history}>
        <SingleFI {...props} />
      </Router>
    );
    expect(wrapper.find('GisidaWrapperMock').length).toEqual(0);
    expect(wrapper.find('HeaderBreadcrumb').length).toEqual(1);
    wrapper.unmount();
  });

  it('It works with the Redux store', () => {
    const mock: any = jest.fn();
    const supersetServiceMock: any = jest.fn(async () => []);
    store.dispatch(fetchPlans(fixtures.plans as Plan[]));
    store.dispatch(fetchGoals(fixtures.goals));
    store.dispatch(fetchJurisdictions(fixtures.jurisdictions));
    const props = {
      completeReactivePlansArray: [fixtures.completeReactivePlan],
      completeRoutinePlansArray: [fixtures.completeRoutinePlan],
      currentReactivePlansArray: [fixtures.plan2],
      currentRoutinePlansArray: [fixtures.plan1],
      history,
      jurisdiction: fixtures.jurisdictions[0],
      location: mock,
      match: {
        isExact: true,
        params: { id: fixtures.plan1.id },
        path: `${FI_SINGLE_URL}/:id`,
        url: `${FI_SINGLE_URL}/16`,
      },
      supersetService: supersetServiceMock,
      // tslint:disable-next-line: object-literal-sort-keys
      fetchPlansActionCreator: jest.fn(),
      fetchGoalsActionCreator: jest.fn(),
      fetchJurisdictionActionCreator: jest.fn(),
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedSingleFI {...props} />
        </Router>
      </Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('GisidaWrapperMock').props()).toMatchSnapshot();
    wrapper.unmount();
  });

  it('calls superset with the correct params', () => {
    const mock: any = jest.fn();
    const supersetMock: any = jest.fn();
    supersetMock.mockImplementation(() => Promise.resolve(fixtures.plans));
    const props = {
      history,
      location: mock,
      match: mock,
      supersetService: supersetMock,
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedActiveFocusInvestigation {...props} />
        </Router>
      </Provider>
    );
    expect(supersetMock).toHaveBeenCalledWith(0, {});
    wrapper.unmount();
  });
});
