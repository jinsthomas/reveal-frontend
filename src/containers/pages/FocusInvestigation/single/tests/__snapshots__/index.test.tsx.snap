// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/pages/SingleFI renders SingleFI correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <SingleFI
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    location={[MockFunction]}
    match={
      Object {
        "isExact": true,
        "params": Object {
          "id": "16",
        },
        "path": "/focus-investigation/view/:id",
        "url": "/focus-investigation/view/16",
      }
    }
  >
    <div>
      <h2
        className="page-title mt-4 mb-5"
      >
        Focus Investigation
         in 
        Focus Area 4
      </h2>
      <Row
        tag="div"
      >
        <div
          className="row"
        >
          <Col
            className="col-6"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col-6 col"
            >
              <h4
                className="mb-4"
              >
                Focus Area Information
              </h4>
              <div
                style={
                  Object {
                    "background": "lightgrey",
                    "height": "200px",
                  }
                }
              />
              <dl
                className="row mt-3"
              >
                <dt
                  className="col-5"
                >
                  Province
                </dt>
                <dd
                  className="col-7"
                >
                  Province 4
                </dd>
                <dt
                  className="col-5"
                >
                  District
                </dt>
                <dd
                  className="col-7"
                >
                  District 4
                </dd>
                <dt
                  className="col-5"
                >
                  Canton
                </dt>
                <dd
                  className="col-7"
                >
                  Canton 4
                </dd>
                <dt
                  className="col-5"
                >
                  Village
                </dt>
                <dd
                  className="col-7"
                >
                  Village 4
                </dd>
                <dt
                  className="col-5"
                >
                  FI Status
                </dt>
                <dd
                  className="col-7"
                >
                  B1
                </dd>
                <dt
                  className="col-5"
                >
                  Last Visit
                </dt>
                <dd
                  className="col-7"
                >
                  29/01/19
                </dd>
                <dt
                  className="col-5"
                >
                  FI Response Adherence
                </dt>
                <dd
                  className="col-7"
                >
                  Yes
                </dd>
              </dl>
              <hr />
              <h5
                className="mb-4 mt-4"
              >
                Past Investigations
              </h5>
              <Table
                responsiveTag="div"
                tag="table"
              >
                <table
                  className="table"
                >
                  <tbody>
                    <tr>
                      <td>
                        <a
                          href="#map"
                        >
                          <FontAwesomeIcon
                            border={false}
                            className=""
                            fixedWidth={false}
                            flip={null}
                            icon={
                              Array [
                                "fas",
                                "map",
                              ]
                            }
                            inverse={false}
                            listItem={false}
                            mask={null}
                            pull={null}
                            pulse={false}
                            rotation={null}
                            size={null}
                            spin={false}
                            symbol={false}
                            title=""
                            transform={null}
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-map fa-w-18 "
                              data-icon="map"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 576 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M0 117.66v346.32c0 11.32 11.43 19.06 21.94 14.86L160 416V32L20.12 87.95A32.006 32.006 0 0 0 0 117.66zM192 416l192 64V96L192 32v384zM554.06 33.16L416 96v384l139.88-55.95A31.996 31.996 0 0 0 576 394.34V48.02c0-11.32-11.43-19.06-21.94-14.86z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </FontAwesomeIcon>
                        </a>
                          
                        <a
                          href="/focus-investigation/view/13"
                        >
                          29/01/19
                        </a>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </Table>
            </div>
          </Col>
          <Col
            className="col-6"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col-6 col"
            >
              <div
                className="fi-active"
              >
                <h5
                  className="mb-4 mt-1"
                >
                  <a
                    href="#map"
                  >
                    <FontAwesomeIcon
                      border={false}
                      className=""
                      fixedWidth={false}
                      flip={null}
                      icon={
                        Array [
                          "fas",
                          "map",
                        ]
                      }
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size={null}
                      spin={false}
                      symbol={false}
                      title=""
                      transform={null}
                    >
                      <svg
                        aria-hidden="true"
                        className="svg-inline--fa fa-map fa-w-18 "
                        data-icon="map"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        style={Object {}}
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M0 117.66v346.32c0 11.32 11.43 19.06 21.94 14.86L160 416V32L20.12 87.95A32.006 32.006 0 0 0 0 117.66zM192 416l192 64V96L192 32v384zM554.06 33.16L416 96v384l139.88-55.95A31.996 31.996 0 0 0 576 394.34V48.02c0-11.32-11.43-19.06-21.94-14.86z"
                          fill="currentColor"
                          style={Object {}}
                        />
                      </svg>
                    </FontAwesomeIcon>
                  </a>
                    
                  <a
                    href="/focus-investigation/view/13"
                  >
                    Active Investigation: 
                    29/01/19
                  </a>
                </h5>
                <dl
                  className="row mt-3"
                >
                  <dt
                    className="col-5"
                  >
                    Reveal User
                  </dt>
                  <dd
                    className="col-7"
                  >
                    username
                  </dd>
                  <dt
                    className="col-5"
                  >
                    Case Notification Date
                  </dt>
                  <dd
                    className="col-7"
                  >
                    29/01/19
                  </dd>
                  <dt
                    className="col-5"
                  >
                    Case Classification
                  </dt>
                  <dd
                    className="col-7"
                  >
                    Imported
                  </dd>
                  <dt
                    className="col-5"
                  >
                    1-3-7 Adherence
                  </dt>
                  <dd
                    className="col-7"
                  >
                    &lt;1
                    , 
                    0
                    , 
                    6
                     days to go
                  </dd>
                  <dt
                    className="col-5"
                  >
                    Complete
                  </dt>
                  <dd
                    className="col-7"
                  >
                    No
                  </dd>
                </dl>
                <hr />
                <h5
                  className="mb-4 mt-4"
                >
                  Response
                </h5>
                <div
                  className="responseItem"
                >
                  <p>
                    T&T: Yes
                  </p>
                  <div
                    className="targetItem"
                  >
                    <p>
                      Target: 13 of 50 people within 1 km
                    </p>
                    <ProgressBar
                      decimalPoints={0}
                      height="10px"
                      max={50}
                      min={0}
                      value={13}
                    >
                      <div
                        className="progress"
                        style={
                          Object {
                            "height": "10px",
                            "marginBottom": "15px",
                          }
                        }
                      >
                        <div
                          aria-valuemax={50}
                          aria-valuemin={0}
                          aria-valuenow={26}
                          className="progress-bar"
                          role="progressbar"
                          style={
                            Object {
                              "backgroundColor": "#ff5c33",
                              "width": "26%",
                            }
                          }
                        />
                      </div>
                    </ProgressBar>
                  </div>
                  <div
                    className="targetItem"
                  >
                    <p>
                      Target: 9 of 10 HHs within 1 km
                    </p>
                    <ProgressBar
                      decimalPoints={0}
                      height="10px"
                      max={10}
                      min={0}
                      value={9}
                    >
                      <div
                        className="progress"
                        style={
                          Object {
                            "height": "10px",
                            "marginBottom": "15px",
                          }
                        }
                      >
                        <div
                          aria-valuemax={10}
                          aria-valuemin={0}
                          aria-valuenow={90}
                          className="progress-bar"
                          role="progressbar"
                          style={
                            Object {
                              "backgroundColor": "#33ad33",
                              "width": "90%",
                            }
                          }
                        />
                      </div>
                    </ProgressBar>
                  </div>
                </div>
                <div
                  className="responseItem"
                >
                  <p>
                    ITN: Yes
                  </p>
                  <div
                    className="targetItem"
                  >
                    <p>
                      Target: 8 of 10 Houses
                    </p>
                    <ProgressBar
                      decimalPoints={0}
                      height="10px"
                      max={10}
                      min={0}
                      value={8}
                    >
                      <div
                        className="progress"
                        style={
                          Object {
                            "height": "10px",
                            "marginBottom": "15px",
                          }
                        }
                      >
                        <div
                          aria-valuemax={10}
                          aria-valuemin={0}
                          aria-valuenow={80}
                          className="progress-bar"
                          role="progressbar"
                          style={
                            Object {
                              "backgroundColor": "#ff8533",
                              "width": "80%",
                            }
                          }
                        />
                      </div>
                    </ProgressBar>
                  </div>
                  <div
                    className="targetItem"
                  >
                    <p>
                      Target: 15 of 100%
                    </p>
                    <ProgressBar
                      decimalPoints={0}
                      height="10px"
                      max={100}
                      min={0}
                      value={15}
                    >
                      <div
                        className="progress"
                        style={
                          Object {
                            "height": "10px",
                            "marginBottom": "15px",
                          }
                        }
                      >
                        <div
                          aria-valuemax={100}
                          aria-valuemin={0}
                          aria-valuenow={15}
                          className="progress-bar"
                          role="progressbar"
                          style={
                            Object {
                              "backgroundColor": "#ff3",
                              "width": "15%",
                            }
                          }
                        />
                      </div>
                    </ProgressBar>
                  </div>
                </div>
                <Row
                  className="mt-5"
                  tag="div"
                >
                  <div
                    className="mt-5 row"
                  >
                    <Col
                      className="col-6 offset-md-3"
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <div
                        className="col-6 offset-md-3 col"
                      >
                        <button
                          className="btn btn-outline-primary btn-block"
                          type="button"
                        >
                          Mark as complete
                        </button>
                      </div>
                    </Col>
                  </div>
                </Row>
              </div>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </SingleFI>
</Router>
`;
